### ISSUE TITLE
Complex type info for constants module

### ISSUE THREAD

**Issue by zachriggle** [Gallopsled/pwntools#503]
Currently, the constants module is just that -- a listing of constant values.

However, we lose a bit of semantic value when the user passes in e.g. `shellcraft.syscall(SYS_execve, ...)`.

We could preserve the information if the module instead created lightweight wrappers that, when printed or `str()`'ed, give the name of the constant.

The implementation would be pretty simple as well, mostly just a proxy for an integer which prints a name instead of the integer value.

This would allow us to simply import all of these values into the template space automagically via some means or another.  This would mean we no longer have to pass in by string e.g. `${i386.mov('eax', 'SYS_execve')}`.  All checks for `isinstance(obj, int)` still work.

This effectively removes the need to call into `constants.eval` from within a template.

```py
class Constant(int):
    def __new__(cls, s, i):
        obj = super(Constant, cls).__new__(cls, i)
        obj.s = s
        return obj
    def __str__(self):
        return self.s
    def __repr__(self):
        return 'Constant(%r, %#x)' % (self.s,int(self))
```

```py
>>> SYS_execve = Constant('SYS_execve', 11)
>>> SYS_execve
Constant('SYS_execve', 0xb)
>>> "%s(%i, %i, %i)" % (SYS_execve, 1, 2, 3)
'SYS_execve(1, 2, 3)'
>>> print SYS_execve
SYS_execve
>>> isinstance(SYS_execve, int)
True
>>> repr(SYS_execve)
"Constant('SYS_execve', 0xb)"
>>> hex(int(SYS_execve))
'0xb'
>>> "%#x" % SYS_execve
'0xb'
>>> SYS_execve == 0xb
True
>>> SYS_execve < 13
True
>>> 0 <= SYS_execve < 13
True
>>> SYS_execve < 10
False
>>> -127 < SYS_execve & 0xff < 127
True
>>> SYS_execve ^ 0xff
244
```

**Comment by zachriggle** [comment]
I've implemented this and it works out pretty well.  There's a bunch of modifications that has to be made to the shellcraft templates to remove use of the strings.  It's a bummer that they don't let you do `from pwnlib.constants import *` in Mako templates.

**Comment by Idolf** [comment]
This was fixed by #636.
